src/main/java/org/zalando/riptide/Router.java

```diff
@@ -28,22 +28,55 @@
 import java.io.IOException;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.stream.Stream;
 
-import static java.util.stream.Collectors.toList;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toMap;
 
-final class Router {
+public final class Router<A> {
 
-    final <A, S> Capture route(final ClientHttpResponse response, final List<HttpMessageConverter<?>> converters,
-            final Selector<A> selector, final Collection<Binding<A>> bindings) {
+    private final Selector<A> selector;
+    private final Map<A, Binding<A>> bindings;
 
-        @Nullable
-        final Binding<A> wildcard = findWildcard(bindings);
+    private Router(Selector<A> selector, Map<A, Binding<A>> bindings) {
+        this.selector = selector;
+        this.bindings = bindings;
+    }
+
+    @SafeVarargs
+    public static <A> Router<A> create(final Selector<A> selector, final Binding<A>... bindings) {
+        return create(selector, asList(bindings));
+    }
+
+    public static <A> Router<A> create(final Selector<A> selector, final Collection<Binding<A>> bindings) {
+        return new Router<A>(selector, Router.createMap(bindings));
+    }
+
+    private static <A> Map<A, Binding<A>> createMap(Collection<Binding<A>> bindings) {
```

Comment:
> I'd leave this up to the Selector implementation

The maps are used and generated by the router. So I find this the natural private place.